src/Analyzers/CSharp/Analyzers/PreferTrailingComma/PreferTrailingCommaDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.PreferTrailingComma
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class PreferTrailingCommaDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public PreferTrailingCommaDiagnosticAnalyzer() : base(
+            diagnosticId: IDEDiagnosticIds.PreferTrailingCommaDiagnosticId,
+            enforceOnBuild: EnforceOnBuildValues.PreferTrailingComma,
+            option: CSharpCodeStyleOptions.PreferTrailingComma,
```

Comment:
😢 Every time we add an option it makes me sad.

I've found in the case of this maintainability rule, it also adds to confusion. Some users think this is a readability rule that [should be configurable](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3153). However, it is not a readability rule, and mistaking the goal here will lead some users to incorrectly configure the rule.

Reply:
@sharwell Are you referring to the fact that `csharp_style_prefer_trailing_comma = false` will disable the analyzer rather than enforcing a non-trailing comma? Or in other words, you like this code-style option to be used by code generation but not by the analyzer?

If that's the case, I definitely agree, but https://github.com/dotnet/roslyn/issues/60584 needs a decision.